examples/step-20/doc/intro.dox

```diff
@@ -439,115 +440,162 @@ matrix-vector products.
 
-In this code, the constructor takes a reference to a block sparse matrix for
-the entire system, and a reference to the object representing the inverse of
-the mass matrix. It stores these using <code>SmartPointer</code> objects (see
-step-7), and additionally allocates two temporary vectors <code>tmp1</code> and
-<code>tmp2</code> for the vectors labeled $w,y$ in the list above.
+    PreconditionJacobi<> preconditioner_M;
+    preconditioner_M.initialize(M);
 
-In the matrix-vector multiplication function, the product $Sv$ is performed in
-exactly the order outlined above. Note how we access the blocks $B$ and $B^T$
-by calling <code>system_matrix->block(0,1)</code> and
-<code>system_matrix->block(1,0)</code> respectively, thereby picking out
-individual blocks of the block system. Multiplication by $M^{-1}$ happens
-using the object introduced above.
+    ReductionControl reduction_control_M(2000, 1.0e-18, 1.0e-10);
+    SolverCG<>    solver_M(reduction_control_M);
 
-With all this, we can go ahead and write down the solver we are going to
-use. Essentially, all we need to do is form the right hand sides of the two
-equations defining $P$ and $U$, and then solve them with the Schur complement
-matrix and the mass matrix, respectively:
+    const auto op_M_inv = inverse_operator(op_M, solver_M, preconditioner_M);
+@endcode
+Rather than using a SolverControl we use the ReductionControl class here
+that stops iterations when either an absolute tolerance is reached (for
+which we choose $10^{-18}$) or when the residual is reduced by a certain
+factor (here, $10^{-10}$). In contrast the SolverControl class only checks
+for absolute tolerances. We have to use ReductionControl in our case to
+work around a minor issue: The right hand sides that we  will feed to
+<code>op_M_inv</code> are essentially formed by residuals that naturally
+have vastly differing norms. This makes control by an absolute tolerance
+very error prone.
+
+We now have a LinearOperator <code>op_M_inv</code> that we can use to
+construct more complicated operators such as the Schur complement $S$.
+Assuming that <code>B</code> is a reference to the upper right block
+constructing a LinearOperator <code>op_S</code> is a matter of two lines:
+@code
+    const auto op_B = linear_operator(B);
+    const auto op_S = transpose_operator(op_B) * op_M_inv * op_B;
+@endcode
+Here, the multiplication of three LinearOperator objects yields a composite
+object <code>op_S</code> whose <code>vmult()</code> function first applies
+$B$, then $M^{-1}$ (i.e. solving an equation with $M$), and finally $B^T$
+to any given input vector. In that sense <code>op_S.vmult()</code> is
+similar to the following code:
+@code
+    B.vmult (tmp1, src); // multiply with the top right block: B
+    solver_M(M, tmp2, tmp1, preconditioner_M); // multiply with M^-1
+    B.Tvmult (dst, tmp2); // multiply with the bottom left block: B^T
+@endcode
+(<code>tmp1</code> and <code>tmp2</code> are two temporary vectors).
```

Comment:
I think it's worth saying that in neither of these two cases is the product of matrices ever formed!